lib/record_store/provider/provider_utils/waiter.rb

```diff
@@ -5,7 +5,7 @@ def initialize(message = nil)
 
   attr_accessor :message
 
-  def wait(sleep_time)
+  def wait(sleep_time, message = @message)
```

Comment:
I would recommend a keyword argument instead of positional just to make sure that we know what we're doing:
```suggestion
  def wait(sleep_time, message: @message)
```

That'll require to explicitly mention `message:` when calling, but makes it clearer (at least to me) what we are doing.

<details><summary>One thing I'm wondering about Ruby is if it's evaluating the default values of methods when the method is called or when the class is instantiated. So I tested and this seems to be working. <em>(Click to expand)</em></summary>
<p>

```ruby
irb(main):001:0> class Test
irb(main):002:1>   def initialize
irb(main):003:2>     @val = 2
irb(main):004:2>   end
irb(main):005:1>   attr_accessor :val
irb(main):006:1>   def test(val: @val)
irb(main):007:2>     puts "val is #{val}; @val is #{@val}"
irb(main):008:2>   end
irb(main):009:1> end
=> :test
irb(main):010:0> t = Test.new
=> #<Test:0x00007fa8b0890250 @val=2>
irb(main):011:0> t.test
val is 2; @val is 2
=> nil
irb(main):012:0> t.test(val: 3)
val is 3; @val is 2
=> nil
irb(main):013:0> t.test
val is 2; @val is 2
=> nil
irb(main):014:0> t.val = 4
=> 4
irb(main):015:0> t.test
val is 4; @val is 4
=> nil
irb(main):016:0> t.test(val: 3)
val is 3; @val is 4
=> nil
irb(main):017:0>
```

</p>
</details>

Reply:
Thanks for the suggestion and for investigating! Using a keyword argument sounds better to me too.